### PR TITLE
Improvements to TVsCE 01

### LINKED ISSUE #1004: What will happen if the VsCode user clicks on multiple commands rapidly?
### Describe the feature you would like

As previously mentioned in a code review, our commands:

1. Load config file
2. Make changes
3. Save to config file

If multiple concurrent commands do this at the same time, some writes can undo the effects of previous writes.
On the CLI, this is not as serious as it is in TVsCE, because the user is less likely to do it.

On TVsCE we have the options to:

1. Make sure the CLI can reliably handle concurrent commands
2. Or will gracefully fail the second command if a previous one is running
3. Or will properly queue concurrent commands
4. Or the TVsCE disables other commands (or queues them) while a previous one is running

### Is this feature solving a problem?

Possible unexpected results if the user runs multiple commands.

### Please share any additional details you have?

Details about the problem your feature solves

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #1029: Update VSCE name and description
Minor updates to the VSCE landing page:

- [ ] Update 'taqueria' to Taqueria
- [ ] Update description to 'A better way to build on Tezos'
- [ ] Add the `preview` tag to the extension - https://devblogs.microsoft.com/visualstudio/new-preview-label-for-visual-studio-extensions-2/

### PR DESCRIPTION
**Pull request by AlirezaHaghshenas**
Closes #1029
Closes #1004

### CODE REVIEW

**Comment by AlirezaHaghshenas** [comment]
closes #1004

**Review by russellmorton — APPROVED** [review]

File: taqueria-vscode-extension/src/lib/gui/SandboxesDataProvider.ts [hunk 1]
```diff
@@ -13,39 +14,84 @@ export class SandboxesDataProvider implements vscode.TreeDataProvider<SandboxTre
...
+	}
 
-		if (element) {
+	private getEnvironmentNames(config: Util.TaqifiedDir): string[] {
+		if (!config?.config?.environment) {
 			return [];
 		}
+		let environments = Object.entries(config.config.environment);
+		const defaultEnv = config.config.environment['default'];
+		environments = environments.filter(environment => environment[0] !== 'default');
```
Comment (jchenche) [inline taqueria-vscode-extension/src/lib/gui/SandboxesDataProvider.ts:81]: I feel like it's more readable to do the following:

`const environments = Object.entries(config.config.environment);` (from `let` to `const`)
Then
`const nonDefaultEnv = environments.filter(environment => environment[0] !== 'default');` (from `environments` to `const nonDefaultEnv`)

What do you think?

File: taqueria-vscode-extension/src/lib/gui/SandboxesDataProvider.ts [hunk 2]
```diff
@@ -13,39 +14,84 @@ export class SandboxesDataProvider implements vscode.TreeDataProvider<SandboxTre
...
 		}
+	}
 
-		if (element) {
+	private getEnvironmentNames(config: Util.TaqifiedDir): string[] {
+		if (!config?.config?.environment) {
 			return [];
 		}
+		let environments = Object.entries(config.config.environment);
+		const defaultEnv = config.config.environment['default'];
```
Comment (jchenche) [inline taqueria-vscode-extension/src/lib/gui/SandboxesDataProvider.ts:80]: `defaultEnv` doesn't seem to be used anywhere

File: taqueria-vscode-extension/src/lib/helpers.ts [hunk 3]
```diff
@@ -398,12 +424,20 @@ export class VsCodeHelper {
...
 			if (!pluginName) {
 				pluginName = await this.promptForPluginInstallation(projectDir);
-			}
-			if (!pluginName) {
-				return;
+				if (!pluginName) {
+					return;
+				}
 			}
 			pluginName = await this.getPathForPluginSource(pluginName);
```
Comment (jchenche) [inline taqueria-vscode-extension/src/lib/helpers.ts:431]: Would the following be more readable?
```typescript
const pluginName = pluginInfo?.label || await this.promptForPluginInstallation(projectDir);
if (!pluginName) return;
const pathToPlugin = await this.getPathForPluginSource(pluginName);
```
If you don't think it captures the logic better, then we can leave it. I don't have the full context 😅

Comment (jchenche) [reply taqueria-vscode-extension/src/lib/helpers.ts:431]: Np! It's just that I know @mweichert prefers `const` over `let` (from functional programming), which I usually resonate with. But I am ok with either. 😃

**Review by jchenche — APPROVED** [review]

**Pull request merged**